### PR TITLE
[DDW-584] Implement "In progress" download logs notification

### PR DESCRIPTION
**Pull request by daniloprates**
This PR implements "In progress" download logs notification.

## Todo

* [x] Settings screen
* [x] Loading screen
* [x] Copy and translation

## Testing

It's better to download a large logs file, so it's easy to see the "In Progress" notification.

## Screenshots

progress

Loading

Settings

---

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn run flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn run storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,10 @@ Changelog
 
 ## vNext
 
+### Features
```
Comment (nikolaglumac) [inline CHANGELOG.md]: @daniloprates please be careful when adding items to the CHANGELOG so that they are added into the correct release.

**Comment by nikolaglumac** [comment]
@daniloprates may I kindly ask you to fix the conflicts? 🙏

**Comment by daniloprates** [comment]
I think it'd look better with a spinner in place of `successIcon` and the animated ellipsis. 

image

**Comment by nikolaglumac** [comment]
@daniloprates agreed. Can you make it so?

**Comment by daniloprates** [comment]
Sure!

**Comment by daniloprates** [comment]
> @daniloprates agreed. Can you make it so?

@nikolaglumac none and ready to be reviewed.

**Comment by nikolaglumac** [comment]
@daniloprates seems like we have a flow issue too 🙏

File: source/renderer/app/components/notifications/GenericNotification.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  hasEllipsis?: boolean,
+};
+
+@observer
+export default class GenericNotification extends Component<Props> {
+  static contextTypes = {
+    intl: intlShape.isRequired,
+  };
+
+  constructor(props: Props) {
```
Comment (DominikGuzei) [inline source/renderer/app/components/notifications/GenericNotification.js]: Are you sure this works correctly when only setting up the listeners in the `constructor`?
From what i know the constructor is only called once for each component and not again if it unmounts and then mounts again. You should probably use `componentDidMount` instead

File: source/renderer/app/containers/LoadingPage.js [hunk 3]
```diff
@@ -115,34 +127,42 @@ export default class LoadingPage extends Component<InjectedProps> {
...
+        <GenericNotification
+          id={DOWNLOAD_LOGS_SUCCESS_NOTIFICATION_ID}
+          duration={DOWNLOAD_LOGS_SUCCESS_DURATION}
+          show={stores.uiNotifications.isOpen(
+            DOWNLOAD_LOGS_SUCCESS_NOTIFICATION_ID
+          )}
+          actionToListenAndOpen={actions.profile.downloadLogsSuccess}
+          actionToListenAndClose={actions.profile.downloadLogs}
+          openNotification={actions.notifications.open}
+          closeNotification={actions.notifications.closeActiveNotification}
```
Comment (DominikGuzei) [inline source/renderer/app/containers/LoadingPage.js]: will there ever be any other `openNotification` and `closeNotification` prop values? If not we should not repeat the wire-up boilerplate for each instance 😉

Comment (daniloprates) [reply source/renderer/app/containers/LoadingPage.js]: @DominikGuzei not sure how to make things differently here, as it's a component, not a container. Maybe using context? These props will always be the same, but the component doesn't have access to the actions.

File: source/renderer/app/components/notifications/GenericNotification.js (hunk 2 above)
Comment (daniloprates) [reply source/renderer/app/components/notifications/GenericNotification.js]: Are you sure? It's working pretty well, after unmounting and mounting again.

File: source/renderer/app/containers/LoadingPage.js (hunk 3 above)
Comment (daniloprates) [reply source/renderer/app/containers/LoadingPage.js]: @DominikGuzei @nikolaglumac I don't know any other way of doing this. The `<GenericNotification />` component needs to receive the `open` and `closeActiveNotification` notification actions, but as it's a component, it doesn't have access to the `actions`. It works, but's ugly, as every component using it needs to pass them manually. Any suggestion? Or can we leave it like this?

Comment (daniloprates) [reply source/renderer/app/containers/LoadingPage.js]: Done, created a container that provides these 2 props.

**Pull request merged**